### DISCUSSION TITLE
FHMAX in regression tests

### DISCUSSION THREAD

**Discussion by benjamin-cash** [ufs-community/ufs-weather-model#1559] [Q&A] [answered]
I had a question about the settings in model_configure and nems.configure. nhours_fcst in [model_configure.IN](http://model_configure.in/) and stop_n in [nems.configure.cpld.IN](http://nems.configure.cpld.in/) are both set to @[FHMAX]. According to the documentation [here](https://github.com/ufs-community/ufs-weather-model/wiki/Restarting-the-Coupled-Model) that is correct for the initial run, but on restart stop_n does not increase while nhours_fcst does. It seems like these should not both be set to @[FHMAX], unless there is a change happening on restart in rt.sh that I haven't been able to track down.

**Comment by DeniseWorthen** [comment]
Our restart tests always utilize checkpoint restarts; so the control run will be from hr 0-> hr 24 and a restart written at hour 12. The restart run starts from hr 12 -> hr 24. So the fhmax does not change...we're still forecasting 24 hours from the initial (start) time.

**Reply by benjamin-cash** [reply]
Hi @DeniseWorthen - The thing that is confusing me is the documentation on restarting the model (https://github.com/ufs-community/ufs-weather-model/wiki/Restarting-the-Coupled-Model) shows nhours_fcst = 48 and stop_n = 24, but in the templates both of those values are set to FHMAX. I suspect I'm confused because I don't really understand all of the things rt.sh is doing. :)

**Reply by benjamin-cash** [reply]
Hi @DeniseWorthen - Just wondering if you have any follow up to this. From what I understand there is either an issue in the regression test or the documentation, or the value of FHMAX is being changed somewhere between creating nems.configure and model_configure. Or I'm still confused about your answer. :) I have a number of runs going now that have been restarted so it would be good to be sure I understand what is happening here.

@JessicaMeixner-NOAA maybe this is something you could shed some light on if Denise is busy?

**Comment by DeniseWorthen** [comment, marked as answer]
@benjamin-cash I don't believe ``stop_n`` has any impact unless an additional configuration variable (``write_restart_at_endofrun``) is also set. This forces the creation of a mediator restart file at the end of the run. 

The ``stop_alarm`` is created using ``stop_n``, ``stop_option`` and ``stop_ymd`` in med.F90 and these need to be present because they are not currently protected w/ ``isPresent``, ``isSet`` conditionals. 

In med_phases_restart_mod.F90, the alarm is used to write a restart file, but only if the ``write_restart_at_endofrun`` is also set true (false by default). 

```
       ! Stop Alarm
       call ESMF_ClockGetAlarm(clock, alarmname='alarm_stop', alarm=alarm, rc=rc)
       if (ChkErr(rc,__LINE__,u_FILE_u)) return
       if (ESMF_AlarmIsRinging(alarm, rc=rc) .and. write_restart_at_endofrun) then
          AlarmIsOn = .true.
       else
          AlarmIsOn = .false.
       endif
```

**Reply by benjamin-cash** [reply]
Perfect, thanks @DeniseWorthen !
